### PR TITLE
added keys and wordlist and break down into components

### PR DESCRIPTION
**Pull request by leannezhang**

### CODE REVIEW

File: src/components/Game.jsx [hunk 1]
```diff
@@ -32,7 +32,11 @@ class Game extends Component {
   }
 
   calculateScore(word) {
-    return word.length
+    if (word.length >= 3 && word.length <= 8) {
```
Comment (BennyLy604) [inline src/components/Game.jsx:36]: what if word length is 1?

File: src/components/ScoreBox/WordScoreList.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import TotalScore from './TotalScore'
+const WordScoreList = (props) => {
+  const {words, scores, totalScore}  = props
+
+  const wordsList = words.map(function(word, index) {
+    return (<li key={word.toString()+index}>{word}</li>)
```
Comment (BennyLy604) [inline src/components/ScoreBox/WordScoreList.jsx:7]: What's the key used for?

Comment (bosung90) [reply src/components/ScoreBox/WordScoreList.jsx:7]: in React, you need to specify unique key for each component if your component is part of an array of the children props

Comment (BennyLy604) [reply src/components/ScoreBox/WordScoreList.jsx:7]: Yeah that makes sense but this is an `<li>` element which renders as html.  Do we need key?

Comment (leannezhang) [reply src/components/ScoreBox/WordScoreList.jsx:7]: yes. There will be warnings if you dont add a key.
https://facebook.github.io/react/docs/lists-and-keys.html

Comment (bosung90) [reply src/components/ScoreBox/WordScoreList.jsx:7]: yes because 
```
<div className="words">
    <h2>WORD</h2>
    {wordsList}
</div>
```
so that wordsList will be array and requires key or it will give you lots of warnings.

It won't make any difference in performance because of the way key is generated (using index)

File: src/components/Game.jsx (hunk 1 above)
Comment (bosung90) [reply src/components/Game.jsx:36]: I think in boggle you need word length of at least 3

File: src/components/ScoreBox/WordScoreList.jsx (hunk 2 above)
Comment (BennyLy604) [reply src/components/ScoreBox/WordScoreList.jsx:7]: Ah I see.  Do we need word.toString() there?  Can we just use index instead?

File: src/components/Game.jsx (hunk 1 above)
Comment (BennyLy604) [reply src/components/Game.jsx:36]: Yes but for a more robust function, I think a safeguard on this function would be better.  If unit tests were made for this function, it would also need to handle edge cases.  There might be places in the code where the word length is checked but code changes all the time.

Comment (leannezhang) [reply src/components/Game.jsx:36]: Im still working on this. Will fix when the word length is 1. The score should be 0.

File: src/components/ScoreBox/WordScoreList.jsx (hunk 2 above)
Comment (leannezhang) [reply src/components/ScoreBox/WordScoreList.jsx:7]: The documentation said. "When you don't have stable IDs for rendered items, you may use the item index as a key as a last resort."

Comment (BennyLy604) [reply src/components/ScoreBox/WordScoreList.jsx:7]: Yeah the words are unique.  We can either use word.toString() or index.  Using both is probably unnecessary.

Comment (leannezhang) [reply src/components/ScoreBox/WordScoreList.jsx:7]: Index is okay. Just use word.toString() will result an error because it's not unique!
